src/style/palette/index.js

```diff
@@ -0,0 +1,28 @@
+import tint from '../utils/tint';
+import shade from '../utils/shade';
+
+const cachedFunc = cb => (cache = {}) => (weight) => {
+  if (cache[weight]) { return cache[weight]; }
+  const color = cb(weight);
+  cache[weight] = color;
+  return color;
+};
+
+const palette = (config) => {
```

Comment:
some documentation on this would be useful - just so other developers can easily see what it is for and how to use it